fvm/env.go

```diff
@@ -1,1143 +1,13 @@
-	return e.txIndex
-}
-
-func (e *transactionEnv) TxID() flow.Identifier {
-	return e.txID
-}
-
-func (e *transactionEnv) GetComputationLimit() uint64 {
-	return e.tx.GasLimit
-}
-
-func (e *transactionEnv) CreateAccount(env *hostEnv, payer runtime.Address) (address runtime.Address, err error) {
-	flowAddress, err := e.addressGenerator.NextAddress()
-	if err != nil {
-		return address, err
-	}
-
-	err = e.accounts.Create(nil, flowAddress)
-	if err != nil {
-		return address, fmt.Errorf("creating account failed: %w", err)
-	}
-
-	if e.ctx.ServiceAccountEnabled {
-		// uses `FlowServiceAccount.setupNewAccount` from https://github.com/onflow/flow-core-contracts/blob/master/contracts/FlowServiceAccount.cdc
-		invoker := NewTransactionContractFunctionInvocator(
-			common.AddressLocation{Address: common.BytesToAddress(e.ctx.Chain.ServiceAddress().Bytes()), Name: flowServiceAccountContract},
-			"setupNewAccount",
-			[]interpreter.Value{
-				interpreter.NewAddressValue(common.BytesToAddress(flowAddress.Bytes())),
-				interpreter.NewAddressValue(common.BytesToAddress(payer.Bytes())),
-			},
-			[]sema.Type{
-				sema.AuthAccountType,
-				sema.AuthAccountType,
-			},
-			e.ctx.Logger,
-		)
-
-		_, invokeErr := invoker.Invoke(env, e.traceSpan)
-
-		if invokeErr != nil {
-			return address, errors.HandleRuntimeError(invokeErr)
-		}
-	}
-
-	e.ctx.Metrics.RuntimeSetNumberOfAccounts(e.addressGenerator.AddressCount())
-	return runtime.Address(flowAddress), nil
-}
-
-func (e *transactionEnv) isAuthorizerServiceAccount() bool {
-	return e.isAuthorizer(runtime.Address(e.ctx.Chain.ServiceAddress()))
-}
-
-func (e *transactionEnv) isAuthorizer(address runtime.Address) bool {
-	for _, accountAddress := range e.GetSigningAccounts() {
-		if accountAddress == address {
-			return true
-		}
-	}
-	return false
-}
-
-func (e *transactionEnv) UpdateAccountContractCode(address runtime.Address, name string, code []byte) (err error) {
-	return e.contracts.SetContract(address, name, code, e.GetSigningAccounts())
-}
-
-func (e *transactionEnv) RemoveAccountContractCode(address runtime.Address, name string) (err error) {
-	return e.contracts.RemoveContract(address, name, e.GetSigningAccounts())
-}
-
-// AddEncodedAccountKey adds an encoded public key to an existing account.
-//
-// This function returns an error if the specified account does not exist or
-// if the key insertion fails.
-func (e *transactionEnv) AddEncodedAccountKey(address runtime.Address, encodedPublicKey []byte) (err error) {
-	return e.accountKeys.AddEncodedAccountKey(address, encodedPublicKey)
-}
-
-// RemoveAccountKey revokes a public key by index from an existing account.
-//
-// This function returns an error if the specified account does not exist, the
-// provided key is invalid, or if key revoking fails.
-func (e *transactionEnv) RemoveAccountKey(address runtime.Address, keyIndex int) (encodedPublicKey []byte, err error) {
-	return e.accountKeys.RemoveAccountKey(address, keyIndex)
-}
```

Comment:
Methods like this are merged into the main function with the same name so we don't do branching based on the condition `transactionEnv != nil`